plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -462,7 +462,7 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
         DiscretePredicates discretePredicates = null;
         if (!partitionSourceIds.isEmpty()) {
             // Extract identity partition columns
-            Map<Integer, IcebergColumnHandle> columns = getColumns(icebergTable.schema(), typeManager).stream()
+            Map<Integer, IcebergColumnHandle> columns = getColumns(SchemaParser.fromJson(table.getTableSchemaJson()), typeManager).stream()
```

Comment:
Why not `icebergTable.schema()`?
Invoking parser looks expensive, and `icebergTable.schema()` **may** have this already calculated.

Note that we're using other table's attributes
```
Set<Integer> partitionSourceIds = identityPartitionColumnsInAllSpecs(icebergTable);
```

so this methods still needs to deal with "current state of the table" vs "table handle's version" vs "history".
and this is why it seems to me to be OK to call ``icebergTable.schema()`` here

Reply:
I agree with the above said. I dropped the commit.